### PR TITLE
Add selectFromList() in VField

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

**Review by h-haddad — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 1]
```diff
@@ -1929,6 +1927,12 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
       throw InconsistencyException()
     }
   }
+  /**
+   * Returns the list table.
+   */
+  private fun evalListTable_(): Table {
+    return  block!!.executeObjectTrigger(list!!.table) as Table
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1934]: it is a workaround here , already checked

File: galite-core/src/main/kotlin/org/kopi/galite/list/VListColumn.kt [hunk 2]
```diff
@@ -27,7 +28,8 @@ abstract class VListColumn(
   val column: String?,
   private val align: Int,
   val width: Int,
-  val isSortAscending: Boolean
+  val isSortAscending: Boolean,
+  val column_ : Column<*>? = null
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/list/VListColumn.kt:32]: already checked with him, everything is fine

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 3]
```diff
@@ -1850,38 +1843,37 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
+
+        else -> throw InconsistencyException("FATAL ERROR: bad search code: $options")
       }
-      append(" ORDER BY 1")
     }
-    val result = displayQueryList(qrybuf, list!!.columns)
+
+    val query = evalListTable_().slice(columns).select(getSearchCondition_(expression!!)!!).orderBy(list!!.getColumn(0).column_!!)
+
+    val result = displayQueryList(query.toString(), list!!.columns)
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1876]: `displayQueryList_(query,....)`

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 4]
```diff
@@ -633,15 +636,8 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
    * Returns the column name in the table with specified correlation.
    * returns null if the field has no access to this table.
    */
-  @Deprecated("use lookupColumn(corr: Table)")
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:636]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 5]
```diff
@@ -633,15 +636,8 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
   fun lookupColumn(corr: Int): Column<*>? = columns!!.find { corr == it!!.getTable() }?.column
 
-  /**
-   * Returns the column name in the table with specified correlation.
-   * returns null if the field has no access to this table.
-   */
-  fun lookupColumn(corr: Table): Column<*>? = columns!!.find { corr == it!!.getTable_() }?.column
-
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:644]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 6]
```diff
@@ -1093,7 +1087,6 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
    * @param     query           the query holding the tuple
    * @param     column          the index of the column in the tuple
    */
-  @Deprecated("use retrieveQuery_(result: ResultRow, column: Column<*>)")
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1096]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 7]
```diff
@@ -1079,7 +1074,6 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
    * @param     query           the query holding the tuple
    * @param     column          the index of the column in the tuple
    */
-  @Deprecated("use setQuery_(record: Int, query: ResultRow, column: Column<*>)")
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1082]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 8]
```diff
@@ -1064,7 +1060,6 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
    * @param     query           the query holding the tuple
    * @param     column          the index of the column in the tuple
    */
-  @Deprecated("use setQuery_(query: ResultRow, column: Column<*>)")
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1067]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 9]
```diff
@@ -57,6 +59,7 @@ import org.kopi.galite.visual.VExecFailedException
 import org.kopi.galite.visual.VRuntimeException
 import org.kopi.galite.visual.VlibProperties
 import org.kopi.galite.visual.VModel
+import java.sql.SQLException
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:62]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 10]
```diff
@@ -20,7 +20,6 @@ package org.kopi.galite.form
 
 import java.awt.Color
 import java.io.InputStream
-import java.sql.SQLException
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:23]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 11]
```diff
@@ -1817,42 +1819,44 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
     }
   }
 
+  /**
+   * Checks that field value exists in list
+   */
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1824]: remove

**Review by hfazai — APPROVED** [review]

**Review by h-haddad — APPROVED** [review]

**Pull request merged**